### PR TITLE
tests: logging: add log_blocking testsuite to demonstrate bugfix

### PR DESCRIPTION
**Pull request by cfriedt**
Add a testsuite to demonstrate proper functionality of blocking in the log processing thread, and also to document what pitfalls to avoid when blocking in the log processing thread.

### CODE REVIEW

**Comment by cfriedt** [comment]
Probably will want to restructure this with ZStress and do all timings in one testsuite rather than 4 separate builds.

This won't be part of 3.5.0, but I hope the fix is (in a separate PR).

**Review by jgl-meta — DISMISSED** [review]

**Review by nordic-krch — DISMISSED** [review]

**Comment by cfriedt** [comment]
@yperess - Was wondering if you could offer some advice on this pr - could it be improved with zstress? It's a bit unclear to me how that should be set up

File: tests/subsys/logging/log_blocking/testcase.yaml
Comment (yperess) [inline tests/subsys/logging/log_blocking/testcase.yaml:24]: @cfriedt  we currently have `ZTEST_EXPECT_FAIL(<suite_name>, <test_name>)` to support expecting a test case to fail.

Comment (cfriedt) [reply tests/subsys/logging/log_blocking/testcase.yaml:24]: Fixed!

**Comment by cfriedt** [comment]
@jgl-meta, @nordic-krch, @yperess - just a ping. This will likely be all green now, and I updated the test case to expect failure in the case of being stalled.

**Comment by cfriedt** [comment]
* corrected yaml indentation

**Comment by cfriedt** [comment]
@jgl-meta, @nordic-krch, @yperess - should be ready for you.

Technically, this could be backported to a bunch of releases as well.

**Review by nordic-krch — APPROVED** [review]

**Review by jgl-meta — APPROVED** [review]

File: tests/subsys/logging/log_blocking/REAME.md
Comment (cfriedt) [reply tests/subsys/logging/log_blocking/REAME.md:34]: Heh... maybe a follow up PR (feel free)? It might not be worth trying to get another round of approvals.

GitHub has a few shortcomings: 
* previous approvals are invalidated (maybe this isn't the case when adding subsequent fixup commits?)
* cannot explicitly hand off a PR to someone else (afaik, e.g. #65769)

If it were possible to not need to get re-approvals if only fixup commits were added, then that should probably be the norm, and then make the default merge button "squash and merge".

**Comment by henrikbrixandersen** [comment]
How come addition of a test suite was tagged for backporting?

**Pull request merged**
